### PR TITLE
Upgrade xscl 4.3 and fix bugs(have merge conflicts)

### PR DESCRIPTION
**Pull request by EmmaZhu**
Upgrade storage library to 4.3 and fix storage cmdlets bugs to refine error message and let New-AzureStorageContext cmdlet to get endpoint info from current subscription environment.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@EmmaZhu__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (emmazhu). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by stankovski** [comment]
Started on-demand build - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/38/
Started MSI build - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/363/

File: src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj [hunk 1]
```diff
@@ -93,6 +93,18 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Resources.2.18.1-preview\lib\net40\Microsoft.Azure.ResourceManager.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Data.Edm, Version=5.6.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (stankovski) [inline src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj]: You have added new DLLs to the compute library. Please regenerate WIX file to ensure the install is in sync.

**Comment by stankovski** [comment]
Started another MSI job - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/383/

**Comment by stankovski** [comment]
Started another on-demand build - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/48/

**Comment by stankovski** [comment]
@EmmaZhu looks like some storage tests are failing - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/38/artifact/src/Publish/TestResults/StorageDebug.trx

**Comment by yugangw-msft** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/76/
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/404/

**Comment by EmmaZhu** [comment]
I will send another PR include this and some new features.

**Pull request closed without merging**